### PR TITLE
Bonemeal Plant Growth Update

### PR DESCRIPTION
**Pull request by SamJBarney**
* Fixed Crops growing to maturity with only 1 bonemeal
* Trees now go through growth stages with bonemeal, as they do in Vanilla Minecraft

### CODE REVIEW

**Comment by SphinxC0re** [comment]
Test

File: src/World.cpp [hunk 1]
```diff
@@ -1481,10 +1492,19 @@ bool cWorld::GrowRipePlant(int a_BlockX, int a_BlockY, int a_BlockZ, bool a_IsBy
...
 			{
-				FastSetBlock(a_BlockX, a_BlockY, a_BlockZ, BlockType, 7);
+				if (!a_IsByBonemeal)
+				{
+					++BlockMeta;
+				}
+				else
+				{
+					BlockMeta += random.NextInt(4) + 2;
+					BlockMeta = std::min(BlockMeta, (NIBBLETYPE)7);
```
Comment (worktycho) [inline src/World.cpp]: c style cast

File: src/World.cpp [hunk 2]
```diff
@@ -1495,7 +1515,17 @@ bool cWorld::GrowRipePlant(int a_BlockX, int a_BlockY, int a_BlockZ, bool a_IsBy
...
-				FastSetBlock(a_BlockX, a_BlockY, a_BlockZ, BlockType, 7);
+
+				if (!a_IsByBonemeal)
+				{
+					++BlockMeta;
+				}
+				else
+				{
+					BlockMeta += random.NextInt(4) + 2;
+					BlockMeta = std::min(BlockMeta, (NIBBLETYPE)7);
```
Comment (worktycho) [inline src/World.cpp]: c style cast

**Comment by SamJBarney** [comment]
It has been tested. The thing that broke it was I switched to using std::max instead of the ternary operator. It compiled fine on my machine, but travis required a cast to be there.
Also, I just had to move a few braces to match the style.

File: src/World.cpp (hunk 2 above)
Comment (SamJBarney) [inline src/World.cpp]: Ok. I wasn't sure if I needed to use the c++ cast templates for basic types. I know std::reinterpret_cast is used for pointer types, but I was unsure about basic types such as ints and unsigned chars.

**Comment by SamJBarney** [comment]
@worktycho  c-casts have been replaced

**Comment by SphinxC0re** [comment]
Looks good. Could you squash the commits onto the current master?

**Comment by SamJBarney** [comment]
Sure

**Pull request closed without merging**